app/generic/proof/proof.py

```diff
@@ -0,0 +1,217 @@
+import logging
+from enum import Enum
+from typing import Literal, Optional
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    IndyPresSpec,
+    IndyProofRequest,
+    V10PresentationSendRequestRequest,
+    V20PresRequestByFormat,
+    V20PresSendRequestRequest,
+    V20PresSpecByFormatRequest,
+)
+from fastapi import APIRouter, Depends
+
+from app.dependencies import agent_selector
+from app.generic.proof.facades.acapy_proof_v1 import ProofsV1
+from app.generic.proof.facades.acapy_proof_v2 import ProofsV2
+from app.generic.proof.models import PresentationExchange
+
+logger = logging.getLogger(__name__)
+
+
+router = APIRouter(prefix="/generic/proof", tags=["proof"])
+
+ProtocolVersion = Literal["1", "2"]
+
+
+class ProofsFacade(Enum):
+    v10 = ProofsV1
+    v20 = ProofsV2
+
+
+@router.post("/send-request")
+async def send_proof_request(
+    connection_id: str,
+    proof_request: IndyProofRequest,
+    protocol_version: Optional[ProtocolVersion] = "1",
```

Comment:
I think there can be value in making the protocol_version required. this way it is more explicit to the user which version they are using.

Reply:
That was my thinking also. One the one hand, it does require a more conscious choice. On the other hand, though, might save some frustration to provide a default one. Not quite sure which way is better, @TimoGlastra